Added model descriptions and configuration files for ForwardTacotron … (#1450)
openvinotoolkit/open_model_zoo: models/public/forward_tacotron/description/forward_tacotron.md

# ForwardTacotron

## Use Case and High-Level Description

ForwardTacotron is model for text to speech task originally trained on PyTorch\*
then converted to ONNX\* format. Forward Tacotron is the model for mel-spectrogramm regression from text.
For details see [paper](https://arxiv.org/pdf/1703.10135.pdf),[paper](https://arxiv.org/pdf/1905.09263.pdf), [repository](https://github.com/as-ideas/ForwardTacotron).

> ### Steps to Reproduce PyTorch to ONNX Conversion

Reviewer: I think it's misleading to just call this "conversion" when it also includes training. You should change it to "training & conversion" and also clarify in the description that this model has been retrained.

Model is provided in ONNX format, which was obtained by the following steps.

1. Clone the original repository
```sh
git clone https://github.com/as-ideas/ForwardTacotron
cd ForwardTacotron
```
2. Checkout the commit that the conversion was tested on:
```sh
git checkout 78789c1aa845057bb2f799e702b1be76bf7defd0
```
3. Follow README.md and train ForwardTacotron model.

> 4. Copy provided scripts `forward_to_onnx.py` to ForwardTacotron root directory.

Reviewer: ```suggestion
4. Copy provided script `forward_to_onnx.py` to ForwardTacotron root directory.
```

5. Create pretrained directory and copy the best model to it.
```sh
mkdir pretrained
cp checkouts/ljspeech_tts.forward/forward_<iteration>k.pyt pretrained

> ```

Reviewer: Why is this step necessary? Your conversion script can read the model from the original path.

Also, ITYM `checkpoints`, not `checkouts`.

6. Run provided script for conversion ForwardTacotron to onnx format
```sh
python3 forward_to_onnx.py --force_cpu --tts_weights pretrained/forward_<iteration>K.pyt
```
Notes:

>    1. By the reason of unsupported operation in ONNX, the model is divided into two parts: `forward_tacotron_duration_prediction.onnx, forward_tacotron_regression.onnx`.

Reviewer: It would be useful to be more specific WRT what is unsupported. E.g., "Since ONNX doesn't support the X operation, ..."

>    2. We stoped training of the Tacotron model in 183K iteration for aligment generation and stoped ForwardTacotron training in 290K iteration.

Reviewer: ```suggestion
   2. We stopped training of the Tacotron model in 183K iteration for alignment generation and stopped ForwardTacotron training in 290K iteration.
```

## ONNX Models

> We provide pretrained models in ONNX format for user convenience.

Reviewer: This section should clearly go before the previous section.

## Specification

| Metric           | Value              |
|------------------|--------------------|
| Type             | Mean Opinion Score |
| GFlops           | -                  |
| MParams          | -                  |

> | Source framework | PyTorch\*          |

Reviewer: 1. This should be broken up into a composite model specification and 2 sub-model specifications. See `action-recognition-0001`, for example.

2. Are you going to fill in GFlops & MParams?

Author: 2. Yes, I am going to do this, but I want to receive this information after model conversion in accuracy checker.

## Accuracy

Subjective

## Performance

> ### ForwardTacotron duration predictor

Author: Done

## Input

Sequence, name: `input_seq`, shape: [1x241], format: [BxC]
where:

   - B - batch size
   - C - number of symbols in sequence (letters or phonemes)

## Output

1. Duration for input symbols, name: `duration`, shape: [1, 241, 1], format [BxCxH]. Contains predicted duration for each of the symbol in sequence.
   - B - batch size
   - C - number of symbols in sequence (letters or phonemes)
   - H - empty dimension.
2. Processed embeddings, name: `embeddings`, shape: [1, 241, 512], format [BxCxH]. Contains processed embeddings for each symbol in sequence.
   - B - batch size
   - C - number of symbols in sequence (letters or phonemes)
   - H - height of the intermediate feature map.
### ForwardTacotron regression

## Input

> Processed embeddigs aligned by durations, name: `data`, shape: [1x805x512], format: [BxTxC]

Reviewer: ```suggestion
Processed embedding aligned by durations, name: `data`, shape: [1x805x512], format: [BxTxC]
```

Author: I think that embeddings is correct.  There is an example in page 2: https://arxiv.org/pdf/1703.10135.pdf

Reviewer: Sorry, I mistyped it myself. :) You still have a missing "n", though.

where:

   - B - batch size

>    - T - time in mel-spectrogramm

Reviewer: ```suggestion
   - T - time in mel-spectrogram
```

>    - C - processed embedding dimention

Reviewer: ```suggestion
   - C - processed embedding dimension
```

## Output

> Mel-spectorgramm, name: `mel`, shape: [80x805], format: [CxT]

Reviewer: ```suggestion
Mel-spectrogram, name: `mel`, shape: [80x805], format: [CxT]
```

where:

>    - T - time in mel-spectrogramm
>    - C - number of mels in mel-spectrogramm

Reviewer: ```suggestion
   - T - time in mel-spectrogram
   - C - number of mels in mel-spectrogram
```

## Legal Information

The original model is distributed under the following

> [license](https://github.com/as-ideas/ForwardTacotron/blob/78789c1aa845057bb2f799e702b1be76bf7defd0/LICENSE)

Reviewer: ```suggestion
[license](https://github.com/as-ideas/ForwardTacotron/blob/78789c1aa845057bb2f799e702b1be76bf7defd0/LICENSE):
```
